mobile/src/helpers/health.ts

```diff
@@ -0,0 +1,184 @@
+      AppleHealthKit.Constants.Permissions.Water,
+      AppleHealthKit.Constants.Permissions.Zinc,
+      // Respiractory
+      AppleHealthKit.Constants.Permissions.OxygenSaturation,
+      AppleHealthKit.Constants.Permissions.RespiratoryRate,
+      // Sleep
+      AppleHealthKit.Constants.Permissions.SleepAnalysis,
+    ],
+    write: [],
+  },
+};
+
+export const auth = async () => {
+  var isAuth;
+
+  return await GoogleFit.checkIsAuthorized().then(() => {
+    isAuth = GoogleFit.isAuthorized;
+    logger.log('GOOGLE_FIT_IS_AUTHORIZED?', isAuth)
+    isAuth = GoogleFit.authorize(runOptions)
+      .then(authResult => {
+        if (authResult.success) {
+          logger.log('GOOGLE_FIT_IS_AUTHORIZED', authResult.success);
+          return true;
+
+        } else {
+          logger.log('GOOGLEFIT_AUTH_DENIED' + authResult);
+          return false;
+        }
+      })
+      .catch(() => {
+        return false;
+      })
+    return isAuth;
+  })
+}
+
+export const disconnectAndroid = () => { GoogleFit.disconnect() }
+
+export const initHealthKit = async (): Promise<boolean> => {
+  return new Promise((resolve, reject) => {
+    AppleHealthKit.initHealthKit(permissions, (err, results) => {
+      if (err) {
+        console.log('error initializing Healthkit:', err);
+        resolve(false);
+      }
+      console.log('RESULTS FROM INIT HEALTHKIT', results);
+      resolve(true);
+    });
+  })
+}
+
+// returns true if there is step count data that is greater than 0
+export const checkForStepsData = async (): Promise<boolean> => {
+  return new Promise((resolve, reject) => {
+    AppleHealthKit.getStepCount(null, (err, results) => {
+      if (err) {
+        console.log('appleHealthKit.getStepCount error:', err);
+        resolve(false);
+        return;
+      }
+
+      console.log('Step Count results FROM HEALTHKIT', results.value);
+      const res: boolean = results.value && results.value > 0;
+      resolve(res);
+      return;
+    });
+  })
+}
+
+// returns true if there is sleep data
+export const checkForSleepData = async (): Promise<boolean> => {
+  return new Promise((resolve, reject) => {
+    AppleHealthKit.getSleepSamples(null, (err, results) => {
+      if (err) {
+        console.log('appleHealthKit.getSleepSamples error:', err);
+        resolve(false);
+        return;
+      }
+
+      console.log('SleepData results FROM HEALTHKIT', results);
+      const res: boolean = results && results.length > 0;
+      resolve(res);
+      return;
+    });
+  })
+}
+
+export const getDOB = async (): Promise<boolean> => {
+  return new Promise((resolve, reject) => {
+    AppleHealthKit.getDateOfBirth(null, (err, results) => {
+      if (err) {
+        resolve(false);
+        return;
+      }
+
+      console.log('getDateOfBirth FROM HEALTHKIT', results.value);
+      resolve(!!results.value);
+      return
+    });
+  })
+}
+
+export const getAuthStatus = async (): Promise<boolean> => {
```

Comment:
can only retrieve write permissions - read permissions will return 0 if unauthorized (same as if user had no data) - for future reference